### PR TITLE
Improve UI

### LINKED ISSUE #203: patch strange behavior with back button

### LINKED ISSUE #268: Improve UI of Navigation Activity

### PR DESCRIPTION
**Pull request by jlstoupy98**
Change the header of the menu to delete user informations
Delete unused item from menu (General Chat & Start Activity)
Add space in activity description
Add a kind of "dynamic" button in activity description to only have useful button depending on the registering of the user to this activity

### CODE REVIEW

File: app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ public void NavigationTest() {
             Thread.sleep(500);
         }catch(Exception e){}
 
-        onView(withId(R.id.nav_start_activity)).perform(click());
+        /*onView(withId(R.id.nav_start_activity)).perform(click());
```
Comment (RoxaneBurri) [inline app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java:148]: Do you remove the start activity button ?

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java [hunk 2]
```diff
@@ -119,6 +120,30 @@ public void displayDescriptionActivityData() {
...
         getParticipantNames();
         loadActivityHeaderPicture();
+        setButton();
+    }
+
+    /**
+     * Modify the visibility of buttons in the layout
+     */
+    private void setButton() {
+        if (activity.getParticipantId().contains(userId)) {
```
Comment (RoxaneBurri) [inline app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:143]: Nice to add the visibility button !

File: app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java [hunk 3]
```diff
@@ -100,12 +100,12 @@ public static void goToHome(View view) {
      * Starts the chat activity
      * @param view the View in which to start the activity
      */
-    public static void goToChat(View view) {
+    /*public static void goToChat(View view) {
```
Comment (RoxaneBurri) [inline app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java:105]: Why do you comment the goToChat ?

File: app/src/main/res/layout-sw320dp/activity_description.xml [hunk 4]
```diff
@@ -277,7 +277,8 @@
                             android:fontFamily="@font/armata"
                             android:onClick="registerToActivity"
                             android:text="@string/activity_description_register_button"
-                            tools:ignore="ButtonStyle" />
+                            tools:ignore="ButtonStyle"
```
Comment (RoxaneBurri) [inline app/src/main/res/layout-sw320dp/activity_description.xml:355]: Do you add the space around words ?

File: app/src/main/res/layout-sw320dp/activity_home_screen.xml [hunk 5]
```diff
@@ -56,7 +56,7 @@
         android:layout_height="wrap_content"
         android:onClick="signIn"
         android:text="@string/signInButton"
-        android:backgroundTint="@color/red"
+        android:backgroundTint="@color/primary"
```
Comment (RoxaneBurri) [inline app/src/main/res/layout-sw320dp/activity_home_screen.xml:59]: I hope the new color will be nice !

File: app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java (hunk 1 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java:148]: Yes, I remove start activity button and General Chat button from the menu because these buttons only made sense for testing and not for real use

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java (hunk 2 above)
Comment (jlstoupy98) [reply app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:143]: Yes thanks, it's much cleaner to have only the useful and accessible buttons for the user!

File: app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java (hunk 3 above)
Comment (jlstoupy98) [reply app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java:105]: I commented it because this item was removed from the navigation because it was no longer used since the general chat was mainly the first version of the chat and was used for testing.

File: app/src/main/res/layout-sw320dp/activity_description.xml (hunk 4 above)
Comment (jlstoupy98) [reply app/src/main/res/layout-sw320dp/activity_description.xml:280]: Oui, c'est plus propre sur la description de l'activité maintenant !

File: app/src/main/res/layout-sw320dp/activity_home_screen.xml (hunk 5 above)
Comment (jlstoupy98) [reply app/src/main/res/layout-sw320dp/activity_home_screen.xml:59]: Haha, I don't know, we'll have to choose together what we want but in any case now, everything should be very easy to change!

File: app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java [hunk 6]
```diff
@@ -121,6 +123,12 @@ public void NavigationTest() {
         }catch(Exception e){}
         Intents.intended(allOf(hasComponent(MainActivity.class.getName())));
 
+
+        Espresso.pressBack();
```
Comment (Antho1426) [inline app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java]: What is the purpose of warning the user when he presses the back button when he is on the MainMapFragment? Is it conventional (among other apps) to implement this feature?

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java [hunk 7]
```diff
@@ -145,6 +157,19 @@ public void displayDescriptionActivityData() {
...
+                findViewById(R.id.activityRegisterDescription).setVisibility(View.VISIBLE);
+                findViewById(R.id.activityGPSRecDescription).setVisibility(View.GONE);
+                findViewById(R.id.activityChatDescription).setVisibility(View.GONE);
+            } else {
+                findViewById(R.id.activityRegisterDescription).setVisibility(View.VISIBLE);
+                findViewById(R.id.activityRegisterDescription).setEnabled(false);
+                ((TextView) findViewById(R.id.activityRegisterDescription)).setText("No more free places");
+                findViewById(R.id.activityGPSRecDescription).setVisibility(View.GONE);
+                findViewById(R.id.activityChatDescription).setVisibility(View.GONE);
+            }
```
Comment (Antho1426) [inline app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:172]: Why do you use GONE instead of INVISIBLE here? Is it a question of visual aspect and distribution of the buttons on the ActivityDescriptionActivity page?

File: app/src/main/java/com/sdp/movemeet/view/home/HomeScreenActivity.java [hunk 8]
```diff
@@ -47,9 +47,18 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        if (isUserLogged()) startActivity(new Intent(this, MainActivity.class));
-        else
-            startActivity(new Intent(this, LoginActivity.class)); // redirecting the user to the "Login" activity
+        if (isUserLogged()) {
+            Intent intent = new Intent(this, MainActivity.class);// we redirect the user to the "MainActivity"
+            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+            intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
+            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            intent.addCategory(Intent.CATEGORY_HOME);
```
Comment (Antho1426) [inline app/src/main/java/com/sdp/movemeet/view/home/HomeScreenActivity.java:56]: Do those flags clear stacked previous frames? What is the purpose of those flags in the end?

File: app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java [hunk 9]
```diff
@@ -73,6 +73,7 @@ public static void startActivity(View view) {
     public static void goToActivityUpload(View view) {
         Context context = view.getContext();
         Intent intent = new Intent(context, UploadActivityActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
```
Comment (Antho1426) [inline app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java:76]: How can this flag help to have a good back button behavior?

File: app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java [hunk 10]
```diff
@@ -185,25 +189,26 @@ public void onComplete(@NonNull Task<DocumentSnapshot> task) {
...
 
-        View hView = navigationView.inflateHeaderView(R.layout.header);
+        //View hView = navigationView.inflateHeaderView(R.layout.header);
+        navigationView.inflateHeaderView(R.layout.header);
 
-        TextView fullName = hView.findViewById(R.id.text_view_profile_name);
+        /*TextView fullName = hView.findViewById(R.id.text_view_profile_name);
         TextView phone = hView.findViewById(R.id.text_view_profile_phone);
-        TextView email = hView.findViewById(R.id.text_view_profile_email);
+        TextView email = hView.findViewById(R.id.text_view_profile_email);*/
```
Comment (Antho1426) [inline app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java]: Why don't you simply remove those lines since we plan not to print the user info in the drawer anymore?

File: app/src/main/res/layout-sw320dp/header.xml [hunk 11]
```diff
@@ -2,67 +2,31 @@
...
-        android:text="@string/header_phone"
-        android:textAlignment="viewStart"
-        android:textSize="14sp"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintStart_toEndOf="@+id/imageView6"
-        app:layout_constraintTop_toTopOf="parent"
-        app:layout_constraintVertical_bias="0.648">
-
-    </TextView>
+        app:fontFamily="@font/aclonica"
```
Comment (Antho1426) [inline app/src/main/res/layout-sw320dp/header.xml:19]: Didn't we use the "armata" font in the beginning, why do you want to change this?

File: app/src/main/res/values/colors.xml [hunk 12]
```diff
@@ -1,19 +1,40 @@
...
+    <color name="primary_light">#FFCCBC</color>
+    <color name="accent">#CDDC39</color>
+    <color name="primary_text">#212121</color>
+    <color name="secondary_text">#757575</color>
+    <color name="icons">#FFFFFF</color>
+    <color name="divider">#BDBDBD</color>-->
+
+    <!-- test -->
+    <!--<color name="primary">#5794B7</color>-->
+    <color name="primary">#01b2FA</color>
```
Comment (Antho1426) [inline app/src/main/res/values/colors.xml:12]: Why did you rename the colors this way? Is it by convenience to then change the values of the hexadecimal color values!

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -124,17 +123,17 @@
 
     <!-- activity description -->
     <string name="activity_description_title">Title</string>
-    <string name="activity_description_by">By </string>
-    <string name="activity_description_with"> with </string>
-    <string name="activity_description_during"> during </string>
-    <string name="activity_description_hours"> hours </string>
+    <string name="activity_description_by">By &#160;</string>
```
Comment (Antho1426) [inline app/src/main/res/values/strings.xml:126]: What are those signs such as "&#160"? Are they equivalent to a space?

**Review by Antho1426 — APPROVED** [review]

File: app/src/main/res/layout-sw320dp/activity_description.xml (hunk 4 above)
Comment (jlstoupy98) [reply app/src/main/res/layout-sw320dp/activity_description.xml:355]: Yes, it's cleaner on the activity description now!

**Review by RoxaneBurri — APPROVED** [review]

File: app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java (hunk 6 above)
Comment (jlstoupy98) [reply app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java]: I've seen this on a lot of apps, so I thought it would be nice. It keeps you from quitting unintentionally. Thanks to this, it also gives an example of the use of "Dialogs" and "onPressBack()".

File: app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java (hunk 7 above)
Comment (jlstoupy98) [reply app/src/main/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivity.java:172]: Exactly. In fact, both settings make the button disappear visually but there is a difference which is the space allocation in the layout. With "invisible", the button is effectively invisible but its place is still reserved on the layout, whereas with "gone", the place is no longer reserved.

File: app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java (hunk 9 above)
Comment (jlstoupy98) [reply app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java:76]: Exactly. This prevents these activities from being recorded on the activity stack. As soon as you change the activity it is automatically destroyed. Only the MainActivity is persistent and thanks to this, on all the other pages, when we press the back button, we are brought back to the MainActivity

File: app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java (hunk 10 above)
Comment (jlstoupy98) [reply app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java]: That's right, I will delete all my commented lines before merge!

File: app/src/main/res/layout-sw320dp/header.xml (hunk 11 above)
Comment (jlstoupy98) [reply app/src/main/res/layout-sw320dp/header.xml:19]: Yes, but here it's only for the text "MOVEMEET" in the header and it's the font that we already used on the MainActivity

Comment (jlstoupy98) [reply app/src/main/res/layout-sw320dp/header.xml:19]: Yes, but here it's only for the text "MOVEMEET" in the header and it's the font that we already used on the MainActivity

File: app/src/main/res/values/colors.xml (hunk 12 above)
Comment (jlstoupy98) [reply app/src/main/res/values/colors.xml:12]: Yes, it is also more common to name colors in this way because it corresponds more to a color palette. And this allows you to adapt the colors more easily everywhere by simply changing these values.

File: app/src/main/res/values/strings.xml (hunk 13 above)
Comment (jlstoupy98) [reply app/src/main/res/values/strings.xml:126]: Yes it is a space but coded like that, it is taken into account during the display whereas a simple space at the end of the string is not taken into account.
